### PR TITLE
add connection.requested_server_name attribute (SNI)

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
Will use Envoy's Network::Connection::requestedServerName() method

### CODE REVIEW

File: mixer/v1/global_dictionary.yaml [hunk 1]
```diff
@@ -186,6 +186,7 @@
 - connection.sent.bytes
 - connection.sent.bytes_total
 - connection.duration
+- connection.requested_server_name
```
Comment (douglas-reid) [inline mixer/v1/global_dictionary.yaml]: unfortunately, this file is append-only. Please add this to the bottom of the file.

**Comment by vadimeisenbergibm** [comment]
@douglas-reid Thanks, I have moved it to the end.

**Comment by andraxylia** [comment]
@costinm

**Comment by hklai** [comment]
@vadimeisenbergibm 
Please make sure istio/api SHA is updated in both istio/proxy and istio/istio.

**Pull request merged**